DashSync/shared/Categories/NSData/NSData+Dash.m

```diff
@@ -1665,6 +1664,16 @@ - (void)saveToFile:(NSString *)file inDirectory:(NSSearchPathDirectory)directory
     [self writeToFile:dataPath atomically:YES];
 }
 
+- (BOOL)isZeroBytes {
+    const uint8_t *bytes = self.bytes;
+    NSUInteger length = self.length;
+    for (NSUInteger i = 0; i < length; i++) {
+        if (bytes[i] != 0) {
+            return NO;
```

Comment:
We can have a small performance improvement here to make sure we don't have branching:

```
uint8_t sum = 0;
const uint8_t *bytes = self.bytes;
NSUInteger length = self.length;
for (NSUInteger i = 0; i < length; i++) {
    sum += bytes[i];
}

return sum == 0;
```

Reply:
Well, I agree in general with that point, but if the input data consists of 256 bytes, where each byte has a value of 1, the sum will overflow, and the function will return YES, indicating that the data is all zero bytes, which is incorrect.

Reply:
Yeah, my bad, we just need to make `sum` as `uint64_t`